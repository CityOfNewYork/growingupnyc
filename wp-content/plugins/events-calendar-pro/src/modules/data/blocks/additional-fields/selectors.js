/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import identity from 'lodash/identity';
import includes from 'lodash/includes';
import uniq from 'lodash/uniq';

/**
 * Internal dependencies
 */
import { constants } from '@moderntribe/common/data/plugins';

export const getPlugin = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ];
export const getBlocks = createSelector(
	[ getPlugin ],
	( plugin ) => plugin.blocks,
);

export const getAdditionalFields = createSelector(
	[ getBlocks ],
	( blocks ) => blocks.additionalFields,
);

export const getFieldName = ( state, props ) => props.name;

export const getIds = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.allIds,
);

export const getFieldAsObjects = createSelector(
	[ getAdditionalFields ],
	( fields ) => fields.byId,
);

export const getFieldBlock = createSelector(
	[ getFieldAsObjects, getFieldName ],
	( fields, name ) => fields[ name ] || {},
);

export const getFieldDividerList = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerList,
);

export const getFieldDividerEnd = createSelector(
	[ getFieldBlock ],
	( field ) => field.dividerEnd,
);

export const getFieldOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldMetaKey = createSelector(
	[ getFieldBlock ],
	( field ) => field.metaKey,
);

export const getFieldType = createSelector(
	[ getFieldBlock ],
	( field ) => field.type,
);

export const getFieldLabel = createSelector(
	[ getFieldBlock ],
	( field ) => field.label || '',
);

export const getFieldValue = createSelector(
	[ getFieldBlock ],
	( field ) => field.value,
);

export const getTextFieldValue = createSelector(
	[ getFieldValue ],
	( value ) => value || '',
);

export const getTextAreaOutput = createSelector(
	[ getFieldOutput ],
	( output = '' ) => {
		return output.split( '\n' ).filter( identity );
	},
);

export const getFieldIsPristine = createSelector(
	[ getFieldBlock ],
	( field ) => field.isPristine,
);

export const getFieldOptions = createSelector(
	[ getFieldBlock ],
	( field ) => field.options || [],
);

export const getFieldOptionsWithLabels = createSelector(
	[ getFieldOptions ],
	( options ) => {
		return options.map( ( option ) => ( { value: option, label: option } ) );
	},
);

export const getFieldDropdownValue = createSelector(
	[ getFieldBlock ],
	( field ) => ( { value: field.value, label: field.value } ),
);

export const getFieldDropdownOutput = createSelector(
	[ getFieldBlock ],
	( field ) => field.output,
);

export const getFieldCheckboxValue = createSelector(
	[ getTextFieldValue ],
	( value ) => uniq( value.split( '|' ) ),
);

export const getFieldCheckboxOptions = createSelector(
	[ getFieldCheckboxValue, getFieldOptionsWithLabels ],
	( values, optionsWithLabels ) => {
		return optionsWithLabels.map( ( option ) => ( {
			...option,
			isChecked: includes( values, option.value ),
		} ) );
	},
);
