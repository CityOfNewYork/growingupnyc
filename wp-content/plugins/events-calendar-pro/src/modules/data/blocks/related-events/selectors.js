/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getRelatedEventsBlock = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.relatedEvents;

export const getTitle = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.title,
);

export const getDisplayImages = createSelector(
	[ getRelatedEventsBlock ],
	( relatedEvents ) => relatedEvents.displayImages,
);
