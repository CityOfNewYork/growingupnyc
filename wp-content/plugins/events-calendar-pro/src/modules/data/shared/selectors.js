/* eslint-disable max-len */

/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import * as constants from '@moderntribe/events-pro/data/blocks/constants';

export const getRule = rule => rule;

export const getType = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_TYPE ],
);

export const getAllDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_ALL_DAY ],
);

export const getMultiDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MULTI_DAY ],
);

export const getMultiDaySpan = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MULTI_DAY_SPAN ],
);

export const getStartDate = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE ],
);

export const getStartDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE_INPUT ],
);

export const getStartDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_DATE_OBJ ],
);

export const getStartTime = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_TIME ],
);

export const getStartTimeNoSeconds = createSelector(
	[ getStartTime ],
	( startTime ) => startTime.slice( 0, -3 ),
);

export const getStartTimeInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_START_TIME_INPUT ],
);

export const getEndDate = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE ],
);

export const getEndDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE_INPUT ],
);

export const getEndDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_DATE_OBJ ],
);

export const getEndTime = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_TIME ],
);

export const getEndTimeNoSeconds = createSelector(
	[ getEndTime ],
	( endTime ) => endTime.slice( 0, -3 ),
);

export const getEndTimeInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_END_TIME_INPUT ],
);

export const getBetween = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_BETWEEN ],
);

export const getLimitType = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_TYPE ],
);

export const getLimit = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT ],
);

export const getLimitDateInput = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_DATE_INPUT ],
);

export const getLimitDateObj = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_LIMIT_DATE_OBJ ],
);

export const getDays = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_DAYS ],
);

export const getWeek = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_WEEK ],
);

export const getDay = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_DAY ],
);

export const getMonth = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_MONTH ],
);

export const getTimezone = createSelector(
	[ getRule ],
	( rule ) => rule[ constants.KEY_TIMEZONE ],
);
