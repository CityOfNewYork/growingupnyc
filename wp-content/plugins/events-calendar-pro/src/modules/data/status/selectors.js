/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getStatus = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].status;

export const isCompleted = createSelector( getStatus, status => !! status.done );
export const getProgress = createSelector( getStatus, status => status.progress );
