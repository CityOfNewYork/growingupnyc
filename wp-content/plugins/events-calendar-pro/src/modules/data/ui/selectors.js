/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { constants } from '@moderntribe/common/data/plugins';

export const getUI = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].ui;

export const isRepeatBlockVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRepeatBlockVisible
);
export const isRulePanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelVisible
);
export const isExceptionPanelVisible = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelVisible
);
export const isRulePanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isRulePanelExpanded
);
export const isExceptionPanelExpanded = createSelector(
	[ getUI ],
	( ui ) => ui.isExceptionPanelExpanded
);
