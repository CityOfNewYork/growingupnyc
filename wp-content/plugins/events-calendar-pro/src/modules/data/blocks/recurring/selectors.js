/* eslint-disable max-len */

/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	RECURRENCE_TYPE_RULES_OPTIONS,
	SERIES_ENDS_OPTIONS,
} from './options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getRules = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.recurring;
export const getRulesCount = createSelector( getRules, ( rules ) => rules.length );
export const hasRules = createSelector( getRulesCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getRules, getIndex ],
	( rules, index ) => rules[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( RECURRENCE_TYPE_RULES_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
