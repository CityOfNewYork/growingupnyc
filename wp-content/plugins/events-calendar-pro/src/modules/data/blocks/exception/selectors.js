/**
 * External dependencies
 */
import { createSelector } from 'reselect';
import { find } from 'lodash';

/**
 * Internal dependencies
 */
import {
	EXCEPTION_OCCURRENCE_OPTIONS,
} from './options';
import { SERIES_ENDS_OPTIONS } from '@moderntribe/events-pro/data/blocks/recurring/options';
import { constants } from '@moderntribe/common/data/plugins';
import * as selectors from '@moderntribe/events-pro/data/shared/selectors';

export const getExceptions = ( state ) => state[ constants.EVENTS_PRO_PLUGIN ].blocks.exception;
export const getRules = getExceptions;
export const getExceptionsCount = createSelector( getExceptions, ( exceptions ) => exceptions.length );
export const hasExceptions = createSelector( getExceptionsCount, ( count ) => !! count );
export const getIndex = ( _, props ) => props.index;

export const getRule = createSelector(
	[ getExceptions, getIndex ],
	( exceptions, index ) => exceptions[ index ],
);

export const getType = createSelector( getRule, selectors.getType );
export const getAllDay = createSelector( getRule, selectors.getAllDay );
export const getMultiDay = createSelector( getRule, selectors.getMultiDay );
export const getMultiDaySpan = createSelector( getRule, selectors.getMultiDaySpan );
export const getStartDate = createSelector( getRule, selectors.getStartDate );
export const getStartDateObj = createSelector( getRule, selectors.getStartDateObj );
export const getStartDateInput = createSelector( getRule, selectors.getStartDateInput );
export const getStartTime = createSelector( getRule, selectors.getStartTime );
export const getStartTimeNoSeconds = createSelector( getRule, selectors.getStartTimeNoSeconds );
export const getStartTimeInput = createSelector( getRule, selectors.getStartTimeInput );
export const getEndDate = createSelector( getRule, selectors.getEndDate );
export const getEndDateObj = createSelector( getRule, selectors.getEndDateObj );
export const getEndDateInput = createSelector( getRule, selectors.getEndDateInput );
export const getEndTime = createSelector( getRule, selectors.getEndTime );
export const getEndTimeNoSeconds = createSelector( getRule, selectors.getEndTimeNoSeconds );
export const getEndTimeInput = createSelector( getRule, selectors.getEndTimeInput );
export const getBetween = createSelector( getRule, selectors.getBetween );
export const getLimitType = createSelector( getRule, selectors.getLimitType );
export const getLimit = createSelector( getRule, selectors.getLimit );
export const getLimitDateObj = createSelector( getRule, selectors.getLimitDateObj );
export const getLimitDateInput = createSelector( getRule, selectors.getLimitDateInput );
export const getDays = createSelector( getRule, selectors.getDays );
export const getDay = createSelector( getRule, selectors.getDay );
export const getMonth = createSelector( getRule, selectors.getMonth );
export const getWeek = createSelector( getRule, selectors.getWeek );
export const getTimezone = createSelector( getRule, selectors.getTimezone );

export const getTypeOption = createSelector(
	[ getType ],
	( type ) => find( EXCEPTION_OCCURRENCE_OPTIONS, ( option ) => option.value === type ),
);

export const getLimitTypeOption = createSelector(
	[ getLimitType ],
	( limitType ) => find( SERIES_ENDS_OPTIONS, ( option ) => option.value === limitType ),
);
