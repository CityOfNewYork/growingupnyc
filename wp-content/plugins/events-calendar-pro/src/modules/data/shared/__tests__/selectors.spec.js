/**
 * External Dependencies
 */
import * as selectors from '../selectors';

describe( 'shared selectors', () => {
	let state;
	beforeEach( () => {
		state = {
			type: 'single',
			all_day: false,
			multi_day: false,
			multi_day_span: 'next-day',
			start_date: 'October 25, 2018',
			start_time: '07:30:00',
			end_date: 'October 25, 2018',
			end_time: '08:00:00',
			between: 1,
			limit_type: 'count',
			limit: 7,
			days: [],
			week: 'first',
			day: 1,
			month: [],
			timezone: 'UTC-6',
			_limit_date_input: '',
			_limit_date_obj: {},
			_start_date_input: '',
			_start_date_obj: {},
			_end_date_input: '',
			_end_date_obj: {},
			_start_time_input: '07:30',
			_end_time_input: '08:00',
		};
	} );

	const keys = Object.keys( selectors );

	keys.forEach( ( key ) => {
		test( key, () => {
			expect( selectors[ key ]( state ) ).toMatchSnapshot();
		} );
	} );
} );
