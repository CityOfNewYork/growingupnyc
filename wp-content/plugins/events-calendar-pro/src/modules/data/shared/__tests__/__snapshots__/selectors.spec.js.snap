// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shared selectors getAllDay 1`] = `false`;

exports[`shared selectors getBetween 1`] = `1`;

exports[`shared selectors getDay 1`] = `1`;

exports[`shared selectors getDays 1`] = `Array []`;

exports[`shared selectors getEndDate 1`] = `"October 25, 2018"`;

exports[`shared selectors getEndDateInput 1`] = `""`;

exports[`shared selectors getEndDateObj 1`] = `Object {}`;

exports[`shared selectors getEndTime 1`] = `"08:00:00"`;

exports[`shared selectors getEndTimeInput 1`] = `"08:00"`;

exports[`shared selectors getEndTimeNoSeconds 1`] = `"08:00"`;

exports[`shared selectors getLimit 1`] = `7`;

exports[`shared selectors getLimitDateInput 1`] = `""`;

exports[`shared selectors getLimitDateObj 1`] = `Object {}`;

exports[`shared selectors getLimitType 1`] = `"count"`;

exports[`shared selectors getMonth 1`] = `Array []`;

exports[`shared selectors getMultiDay 1`] = `false`;

exports[`shared selectors getMultiDaySpan 1`] = `"next-day"`;

exports[`shared selectors getRule 1`] = `
Object {
  "_end_date_input": "",
  "_end_date_obj": Object {},
  "_end_time_input": "08:00",
  "_limit_date_input": "",
  "_limit_date_obj": Object {},
  "_start_date_input": "",
  "_start_date_obj": Object {},
  "_start_time_input": "07:30",
  "all_day": false,
  "between": 1,
  "day": 1,
  "days": Array [],
  "end_date": "October 25, 2018",
  "end_time": "08:00:00",
  "limit": 7,
  "limit_type": "count",
  "month": Array [],
  "multi_day": false,
  "multi_day_span": "next-day",
  "start_date": "October 25, 2018",
  "start_time": "07:30:00",
  "timezone": "UTC-6",
  "type": "single",
  "week": "first",
}
`;

exports[`shared selectors getStartDate 1`] = `"October 25, 2018"`;

exports[`shared selectors getStartDateInput 1`] = `""`;

exports[`shared selectors getStartDateObj 1`] = `Object {}`;

exports[`shared selectors getStartTime 1`] = `"07:30:00"`;

exports[`shared selectors getStartTimeInput 1`] = `"07:30"`;

exports[`shared selectors getStartTimeNoSeconds 1`] = `"07:30"`;

exports[`shared selectors getTimezone 1`] = `"UTC-6"`;

exports[`shared selectors getType 1`] = `"single"`;

exports[`shared selectors getWeek 1`] = `"first"`;
